coaster/sqlalchemy/statemanager.py

```diff
@@ -525,17 +547,21 @@ def __call__(self, *args, **kwargs):
         transition_before.send(self.obj, transition=self.statetransition)
         # Call the transition function
         try:
-            result = self.statetransition.func(self.obj, *args, **kwargs)
+            self.statetransition.func(self.obj, *args, **kwargs)
+        except AbortTransitionError as e:
+            transition_exception.send(self.obj, transition=self.statetransition, exception=e)
+            return False, str(e)
         except Exception as e:
             transition_exception.send(self.obj, transition=self.statetransition, exception=e)
             raise
+
         # Change the state for each of the state managers
         for statemanager, conditions in self.statetransition.transitions.items():
             if conditions['to'] is not None:  # Allow to=None for the @requires decorator
                 statemanager._set(self.obj, conditions['to'].value)  # Change state
         # Raise a transition-after signal
         transition_after.send(self.obj, transition=self.statetransition)
-        return result
+        return True, self.statetransition.data.get('message', '')
```

Comment:
Whatever is in `data` is project convention. You cannot make _any_ assumptions about its contents here. Just `return result` (as captured above).